mesh/ncmesh.cpp

```diff
@@ -2311,8 +2312,61 @@ int NCMesh::find_local_face(int geom, int a, int b, int c)
    return -1;
 }
 
+
+struct PointMatrixHash
+{
+   std::size_t operator()(const NCMesh::PointMatrix &pm) const
+   {
+      MFEM_ASSERT(sizeof(double) == sizeof(std::uint64_t), "");
+      std::uint64_t hash = 0xf9ca9ba106acbba9;
+      for (int i = 0; i < pm.np; i++)
+      {
+         for (int j = 0; j < pm.points[i].dim; j++)
+         {
+            double coord = pm.points[i].coord[j];
+            hash = 31*hash + *((std::uint64_t*) &coord);
+         }
+      }
+      return hash;
+   }
+};
```

Comment:
Why is this a good hash key?

Reply:
This a variation on the section "Hashing an array of floats" from here: https://cs.stackexchange.com/questions/37952/hash-function-floating-point-inputs-for-genetic-algorithm

I tested this simple version and the original that had more mixing of the bits at the end, but the distribution into the buckets of unordered_map was practically unaffected. The important thing to check is that there aren't any unfortunate collisions (some buckets overfilled), I did that using "DumpBucketSizes" which is left in the struct for potential debugging. 

The hash seems to work OK for our purpose. The initial constant is more or less random (only important if the matrix was empty). The constant 31 is more important, I think it's good that it's a prime. Beyond this and some basic testing I didn't put more thought into this.

Reply:
Some comments or documentation could help a lot here, especially given the "magic number" literals.

Reply:
Added more comments for the hash functor.